### ISSUE TITLE
iOS: Regression in refreshControl on RN 0.64

### ISSUE THREAD

**Issue by kamui545** [facebook/react-native#30912]
## Description

Hello everyone, I might have found a regression on RN 0.64 (I tested every RC, including latest RC3).
The refresh control of every FlatList from my app is randomly showing up "frozen".

This issue seems to be similar to https://github.com/facebook/react-native/issues/7976 which was fixed in the following PR https://github.com/facebook/react-native/pull/9249.

Sadly, the fix was removed in https://github.com/facebook/react-native/pull/28236 and merged to RN 0.64.

I am experiencing this bug in iOS 14, even though in the fix it mentions:
`TODO: Remove when updating to use iOS 10 refreshControl UIScrollView prop.`

## React Native version:

```
System:
    OS: macOS 11.2
    CPU: (8) x64 Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
    Memory: 83.05 MB / 16.00 GB
    Shell: 3.1.2 - /usr/local/bin/fish
  Binaries:
    Node: 14.15.4 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.14.10 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 29, 30
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.3
      System Images: android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6953283
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_282 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: ^17.0.1 => 17.0.1 
    react-native: ^0.64.0-rc.3 => 0.64.0-rc.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce

1. Display screen with FlatList
2. Navigate to other screen
3. Go back to the previous screen with FlatList, you should see the refresh control.

If you start scrolling or pull to refresh, it then disappears.
I am not doing anything fancy and don't have any custom `refreshControl` property on my FlatList.
Even with `refreshing={false}` forced, the refreshControl will show up.

However, I noticed that if I use a refreshControl with a transparent background, the issue is fixed:

```js
refreshControl={
  <RefreshControl
    refreshing={refreshing}
    onRefresh={onRefresh}
    style={{
      backgroundColor : 'transparent', // <---- this somehow fix the issue
    }}
  />
}
```

## Expected Results

Not having any refresh control displayed if it is not refreshing.

## Snack, code example, screenshot, or link to a repository:

I will try to make a snack / repo to reproduce the issue asap.
In the meantime here is a video of the issue:

https://user-images.githubusercontent.com/356978/107233817-46410d00-6a23-11eb-95fe-08d46d50bd2e.mov

**Comment by swrobel** [comment]
Setting `backgroundColor : 'transparent'` did not fix the issue for me, however restoring the following code removed in #28236 did indeed fix it:

```objc
  // Fix for bug #7976
  // TODO: Remove when updating to use iOS 10 refreshControl UIScrollView prop.
  if (self.backgroundColor == nil) {
    self.backgroundColor = [UIColor clearColor];
  }
```

**Comment by swrobel** [comment]
Fixed in https://github.com/facebook/react-native/pull/30978. Mind closing if you can verify that fixes it for you as well, @kamui545?

**Comment by kamui545** [comment]
Just wanted to confirm that the issue seems to be fixed on my end with RN 0.64 stable.

@swrobel Thanks for taking care of this
